### DISCUSSION TITLE
Compilation

### DISCUSSION THREAD

**Discussion by Ruttie2006** [AvaloniaUI/Avalonia#9781] [Q&A]
This isn't really a question about how to implement something in Avalonia, but more a question of how Avaloniais implemented:
I noticed that when using ILSpy to look at an Avalonia project, the following can be found:
![image](https://user-images.githubusercontent.com/80394237/209465718-57445b69-c660-490c-9469-ef2d8574c8e2.png)
My question is, how exactly does Avalonia create this?
I'm curious because I want to implement something like it as well...

**Comment by kekekeks** [comment]
https://github.com/kekekeks/XamlX

**Reply by Ruttie2006** [reply]
I still don't really get how that works either 😓
My main question is I suppose how do you get the compiler to add your IL code to the project?

**Reply by kekekeks** [reply]
We are using Mono.Cecil to patch the binary
